### PR TITLE
add resnet 101/152 variants

### PR DESCRIPTION
**Pull request by brettkoonce**

### CODE REVIEW

**Comment by rxwei** [comment]
I think it would be better to do add more ResNet variants when arrays can be differentiable and when things can be generalized to aggregate types instead of being raw layers. Otherwise this is a lot of code bloat and is making code less readable.

**Comment by brettkoonce** [comment]
kk, what needs to be done to make that happen?

**Comment by rxwei** [comment]
For instance, to reduce code bloat, you can define a struct that includes all the repeating subcomponents. This file is really unnecessarily large right now because of the lack of generalization.

```swift
    var l4l = ResidualIdentityBlock(featureCounts: (1024, 256, 256, 1024))
    var l4m = ResidualIdentityBlock(featureCounts: (1024, 256, 256, 1024))
    var l4n = ResidualIdentityBlock(featureCounts: (1024, 256, 256, 1024))
    var l4o = ResidualIdentityBlock(featureCounts: (1024, 256, 256, 1024))
    var l4p = ResidualIdentityBlock(featureCounts: (1024, 256, 256, 1024))
```

File: ResNet/ResNet50.swift [hunk 1]
```diff
@@ -145,6 +145,33 @@ struct ResidualIdentityBlock: Layer {
     }
 }
 
+struct ResidualIdentityBlockStack: Layer {
```
Comment (brettkoonce) [inline ResNet/ResNet50.swift]: @rxwei let me do some testing/cleanup, but is this pattern closer to what you're after?

File: ResNet/ResNet50.swift [hunk 2]
```diff
@@ -145,6 +145,33 @@ struct ResidualIdentityBlock: Layer {
...
+    var layer1: ResidualIdentityBlock
```
Comment (rxwei) [inline ResNet/ResNet50.swift]: Call it “block1” instead?

**Comment by rxwei** [comment]
Yeah, the improved pattern is what i was expecting. It's much better! I think other existing resnet variants can be generalized this way as well.

File: ResNet/ResNet50.swift (hunk 2 above)
Comment (brettkoonce) [reply ResNet/ResNet50.swift]: okay!  let me do some testing and cleanup and then i will ping you again!

**Pull request merged**